testutil/integration/simnet_test.go

```diff
@@ -37,87 +37,109 @@ import (
 	"github.com/obolnetwork/charon/testutil/beaconmock"
 )
 
+// vcType defines a validator client type.
+type vcType string
+
+// TODO(dhruv): Add support of more VC types to simnet tests.
+const (
+	vcMock vcType = "mock"
+	vcTeku vcType = "teku"
+)
+
 //go:generate go test . -integration -v -run=TestSimnetDuties
 
 func TestSimnetDuties(t *testing.T) {
 	skipIfDisabled(t)
 
 	tests := []struct {
-		name                string
-		scheduledType       core.DutyType
-		duties              []core.DutyType
-		builderAPI          bool
-		builderRegistration bool
-		exit                bool
-		teku                bool
+		name                    string
+		scheduledType           core.DutyType
+		duties                  []core.DutyType
+		vcTypes                 []vcType
```

Comment:
why slice, do we test with multiple types?

Reply:
not as of now, but i'd like to test multiple types in simnet test. 
An argument against testing multiple types in simnet is we already do some multiple types testing in compose smoke tests so these tests would be more or less same other than not involving docker compose files. What do you think?

Reply:
not sure this adds a lot of value here, I'd rather keep it simple for now, single VC. We can do matrix E2E testing.

Reply:
cool, I'll revert this back to single VC.